src/transaction.ts

```diff
@@ -89,6 +90,18 @@ export class StacksTransaction {
     return tx.txid();
   }
 
+  setSignature(signature: string | Buffer): StacksTransaction {
```

Comment:
To avoid confusion with developers thinking this is a mutation, I recommend renaming to `txWithSignature(signature: string | Buffer)`

Reply:
Went with `createTxWithSignature`, is this clearer?

Reply:
looks good 👍